deposit/src/main/java/edu/unc/lib/deposit/normalize/BagIt2N3BagJob.java

```diff
@@ -0,0 +1,222 @@
+			failJob("Unsupported BagIt bag format", "Only filesystem bags are supported.");
+		}
+		
+		// Verify that the bag has all the required parts
+		SimpleResult completeResult = bag.verifyComplete();
+		if (!bag.verifyComplete().isSuccess()) {
+			// Bag did not validate, generate error report and throw exception
+			StringBuilder msg = new StringBuilder();
+			for (String error : completeResult.getErrorMessages()) {
+				msg.append(error).append("\n");
+			}
+			
+			failJob("Unable to normalize bag " + sourcePath + ", it was not complete according to bagit specifications",
+					msg.toString());
+		}
+		
+		Collection<BagFile> payload = bag.getPayload();
+		
+		Property labelProp = dprop(model, DepositRelationship.label);
+		Property hasModelProp = fprop(model, FedoraProperty.hasModel);
+		Property md5sumProp = dprop(model, md5sum);
+		Property locationProp = dprop(model, DepositRelationship.stagingLocation);
+		Resource simpleResource = model.createResource(SIMPLE.getURI().toString());
+		
+		// Turn the bag itself into the top level folder for this deposit
+		com.hp.hpl.jena.rdf.model.Bag bagFolder = model.createBag(new PID("uuid:" + UUID.randomUUID()).getURI());
+		model.add(bagFolder, labelProp, sourceFile.getName());
+		model.add(bagFolder, hasModelProp, model.createResource(CONTAINER.getURI().toString()));
+		top.add(bagFolder);
+		
+		// Add all of the payload objects into the bag folder
+		for (BagFile file : payload) {
+			String filePath = file.getFilepath();
+			
+			Map<Manifest.Algorithm, String> checksums = bag.getChecksums(filePath);
+			
+			Resource fileResource = getFileResource(bagFolder, sourcePath, filePath);
+			
+			// add checksum, size, label
+			String filename = filePath.substring(filePath.lastIndexOf("/") + 1);
+			model.add(fileResource, labelProp, filename);
+			model.add(fileResource, hasModelProp, simpleResource);
+			if (checksums.containsKey(Manifest.Algorithm.MD5)) {
+				model.add(fileResource, md5sumProp, checksums.get(Manifest.Algorithm.MD5));
+			}
+			
+			// Find staged path for the file
+			Path storedPath = Paths.get(sourceFile.getAbsolutePath(), filePath);
+			try {
+				URI stagedURI = stages.getStagedURI(storedPath.toUri());
+				
+				if (stagedURI != null) {
+					model.add(fileResource, locationProp, stagedURI.toString());
+				}
+			} catch (StagingException e) {
+				failJob(e, "Unable to get staged path for file {}", storedPath);
+			}
+			
+		}
+		
+	}
+	
+	private Resource getFileResource(com.hp.hpl.jena.rdf.model.Bag top, String basepath, String filepath) {
+		com.hp.hpl.jena.rdf.model.Bag folderBag = getFolderBag(top, basepath, filepath);
+
+		UUID uuid = UUID.randomUUID();
+		PID pid = new PID("uuid:" + uuid.toString());
+
+		Resource fileResource = top.getModel().createResource(pid.getURI());
+		folderBag.add(fileResource);
+
+		return fileResource;
+	}
+	
+	private com.hp.hpl.jena.rdf.model.Bag getFolderBag(com.hp.hpl.jena.rdf.model.Bag top, String basepath, String filepath) {
```

Comment:
Yeah, that made me slightly sad, but when I was merging our two different versions (lets avoid having to do that in the future) I mostly went with the version I already knew if they looked to do basically the same thing.